### PR TITLE
Backend/decoupling drainer from application

### PR DESCRIPTION
**Pull request by vijaygupta18**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

Making drainer decoupled from application.
Previously in case of addition or deletion of column we had to deploy it multiple times according to our requirement.
After this change it is going to be independent of app.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [x] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-drainer/src/Utils/Utils.hs [hunk 1]
```diff
@@ -149,3 +154,164 @@ shutDownHandler = do
...
+-- this is being used to get the where condition from the json object | Note : INCOMPLETE IMPLEMENTATION
+makeWhereCondition :: Value -> A.Object -> Text
+makeWhereCondition values mappings = case values of
+  A.Object obj -> do
+    let key = HM.keys obj
+    case key of
+      ["$and"] -> getArrayConditionText (fromMaybe A.emptyObject (HM.lookup "$and" obj)) " AND " mappings
+      ["$or"] -> getArrayConditionText (fromMaybe A.emptyObject (HM.lookup "$or" obj)) " OR " mappings
+      -- these conditions need to be implemented safely before using in production
+      ["$gt"] -> quote' (textToSnakeCaseText $ replaceMappings (head key) mappings) <> " > " <> valueToText (fromMaybe A.emptyObject (HM.lookup (head key) obj))
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-drainer/src/Utils/Utils.hs:258]: Can we avoid partial functions like `head`?

File: Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs [hunk 2]
```diff
@@ -1,191 +1,103 @@
...
-          EL.logError ("Create failed: " :: Text) (show cmdsToErrorQueue <> "\n Error: " <> show x :: Text)
-          pure [Left entryIds]
+-- | Generate an insert query for the atlas_driver_offer_bpp schema
+generateInsertForTable :: (Maybe Text, Maybe A.Object, Maybe A.Object) -> Text -> Maybe Text
+generateInsertForTable (mbModel, mbObject, mbMappings) tableName =
+  case (mbModel, mbObject) of
+    (Just _, Just object') -> do
+      let keys = HM.keys object'
+          newKeys = map (`replaceMappings` fromMaybe HM.empty mbMappings) keys
+          newKeys' = map (quote' . textToSnakeCaseText) newKeys
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs:77]: I see that you are using `quietSnake` function for getting table column names. But sometimes its not working, we have exceptions here, we use `mkTableInstancesWithTModifier` for handle this exceptions in Beam instances:

`$(mkTableInstancesWithTModifier ''DriverQuoteT "driver_quote" [("requestId", "search_request_id")])`

Did you handled this in your queries?

Generally, using raw queries looks unsafe. We should discuss this possibility with @hemantmangla and others

File: Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs [hunk 3]
```diff
@@ -1,191 +1,103 @@
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs:1]: Can you remove this pragma?

File: Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs (hunk 2 above)
Comment (vijaygupta18) [reply Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs:77]: here replaceMappings will do that...i'm getting those mappings in strean itself
 I have made a custom instance which will be used to get all these mismatched field mappings and then i send it in streamdata

**Pull request closed without merging**